include/depthai/depthai_wrapper.hpp

```diff
@@ -0,0 +1,57 @@
+#include "depthai/device.hpp"
+#include <string>
+#include <opencv2/opencv.hpp>
+#include <opencv2/core/core.hpp>
+#include <fstream>
+#include <sstream>
+#include "nlohmann/json.hpp"
+#include <vector>
+
+namespace DepthAI
+{
+
+using CV_mat_ptr = std::shared_ptr<cv::Mat>;  // shared ptr for cv::Mat
+
+class DepthAI : public Device
+{
+
+public:
+    /**  Constructor:
+     *   initializing OAK device and creating pipeline
+     *   to capture camera streams and AI outputs.
+     */
+    DepthAI(std::string usb_device, std::string config_file, bool usb2_mode = false);
+
+    /* API to stream output frames from OAK.
+     * AI data and metadata is WIP
+     */
+
+    void get_frames(std::unordered_map<std::string, CV_mat_ptr>& output_streams);
```

Comment:
You should really add more explanation to this function. For example, why do we need a map here? Does it return multiple frames frames in one call? And if the function is called `get_frames`, why is the output parameter called `output_streams`? What are the meanings of key-value pairs inside the map? As a first time reader of your code these comments provide very little info.

Reply:
You did not address this comment

Reply:
Havn't finished wiat. I am testing if that function works first. I still hve to add that. I will tell you to review after that